src/everloop/mod.rs

```diff
@@ -26,37 +28,34 @@ impl<'a> Everloop<'a> {
     /// everloop.set(&vec![matrix_rhal::Rgbw::new(0,0,255,0); 15]);
     /// ```
     pub fn set(&self, leds: &[Rgbw]) {
-        let device_leds = self.bus.device_leds();
+        let device_leds = self.bus.device_leds() as usize;
         if leds.len() > device_leds as usize {
             panic!("Invalid LED set. This device only has {} LEDs", device_leds);
         }
 
         // create write buffer
-        let capacity = device_leds as usize + 2;
-        let mut request: Vec<i32, MAX_LEDS> = Vec::new();
-        request.push(fpga_address::EVERLOOP as i32).unwrap();
-        request.push((device_leds * 4) as i32).unwrap(); // each LED RGBW requires 4 bytes
+        let led_bytes = device_leds * core::mem::size_of::<LedBytes>();
+        let mut request: Vec<u8, MAX_LED_BYTES> = Vec::new();
+        request.extend_from_slice(&(fpga_address::EVERLOOP as i32).to_ne_bytes()).unwrap();
+        request.extend_from_slice(&(led_bytes as i32).to_ne_bytes()).unwrap(); // each LED RGBW requires 4 bytes
 
         // store all LED colors given
         for led in leds {
-            request.push(led.as_bytes()).unwrap();
+            request.extend_from_slice(&led.to_bytes()).unwrap();
```

Comment:
This throws an error when I run it on the Raspberry Pi. 

Example Code:
```
    let bus = hal::bus::init();
    let everloop = hal::Everloop::new(&*bus);
    everloop.set_all(hal::Rgbw::black());
```

Error Log:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: ()', src/everloop/mod.rs:44:13
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
With a Creator, right?  That buffer isn't large enough for the extra header bytes.

Reply:
Yep! I haven't tested it with the Voice yet.